src/App.tsx

```diff
@@ -1,14 +1,53 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
+import './components/App.scss';
 
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [isLoaded, setIsLoaded] = useState(false);
+  const [query, setQuery] = useState('');
+  const [selectFilter, setSelectFilter] = useState('all');
+  const [todo, setTodo] = useState<Todo | null>(null);
+
+  useEffect(() => {
+    const fetchTodos = async () => {
+      setTodos(await getTodos());
+      setIsLoaded(true);
+    };
+
+    fetchTodos();
+  }, []);
+
+  const filterTodos = (initialTodos: Todo[]) => {
+    switch (selectFilter) {
+      case 'all':
+        return initialTodos.filter(actualTodo => actualTodo.title.includes(query));
+      case 'completed':
+        return initialTodos.filter(actualTodo => actualTodo.title.includes(query) && actualTodo.completed);
+      case 'active':
+        return initialTodos.filter(actualTodo => actualTodo.title.includes(query) && !actualTodo.completed);
+      default:
+        return initialTodos;
+    }
```

Comment:
Let's use one filter instead
```suggestion
     initialTodos.filter(actualTodo => {
       switch (selectFilter) {
        case 'all':
          return ...
        case 'completed':
          return ...
        case 'active':
          return ...
        default:
          return ...;
      }
     }
```